routing/checkpoints.hpp

```diff
@@ -40,32 +43,58 @@ class Checkpoints final
     m_points.front() = start;
   }
 
+  m2::PointD const & GetPoint(size_t pointIdx) const
+  {
+    CHECK_LESS(pointIdx, m_points.size(), ());
+    return m_points[pointIdx];
+  }
+
   std::vector<m2::PointD> const & GetPoints() const
   {
     CheckValid();
     return m_points;
   }
 
-  size_t GetArriveIdx() const { return m_arriveIdx; }
+  size_t GetNumSubroutes() const
+  {
+    CheckValid();
+    return m_points.size() - 1;
+  }
+
+  size_t GetArrivedIdx() const { return m_arrivedIdx; }
 
   void ArriveNextPoint()
   {
-    ++m_arriveIdx;
+    ++m_arrivedIdx;
     CheckValid();
   }
 
   void CheckValid() const
   {
     CHECK_GREATER_OR_EQUAL(m_points.size(), 2,
                            ("Checkpoints should at least contain start and finish"));
-    CHECK_LESS(m_arriveIdx, m_points.size(), ());
+    CHECK_LESS(m_arrivedIdx, m_points.size(), ());
   }
 
 private:
   // m_points contains start, finish and intermediate points.
   std::vector<m2::PointD> m_points;
-  // Arrive idx is the index of the checkpoint by which the user passed by.
+  // Arrived idx is the index of the checkpoint by which the user passed by.
   // By default, user has arrived at 0, it is start point.
-  size_t m_arriveIdx = 0;
+  size_t m_arrivedIdx = 0;
 };
+
+inline std::string DebugPrint(Checkpoints const & checkpoints)
+{
+  std::ostringstream out;
+  out.precision(8);
+  out << "Checkpoints(";
+  for (auto const & point : checkpoints.GetPoints())
+  {
+    auto const latlon = MercatorBounds::ToLatLon(point);
+    out << latlon.lat << " " << latlon.lon << ", ";
```

Comment:
But from my point of view, better approach is to use ```fixed``` from iomanip and print exactly 6 digits after the decimal point.